game/graphics/renderer.cpp

```diff
@@ -596,8 +671,23 @@ void Renderer::drawTonemapping(Encoder<CommandBuffer>& cmd) {
   if(mul>0)
     p.mul = mul;
 
-  cmd.setUniforms(*tonemapping.pso, tonemapping.uboTone, &p, sizeof(p));
-  cmd.draw(Resources::fsqVbo());
+  if (settings.cmaa2Enabled) {
+    cmd.setFramebuffer({});
+
+    tonemapping.uboTone.set(2, cmaa2.sceneTonemappedUav);
+    tonemapping.uboTone.set(3, cmaa2.sceneHdrLumaUav);
+    cmd.setUniforms(*tonemapping.computePso, tonemapping.uboTone, &p, sizeof(p));
+
+    const uint32_t threadGroupSize = 8;
+    const uint32_t groupsCountX = (cmaa2.sceneTonemappedUav.w() + threadGroupSize - 1) / threadGroupSize;
+    const uint32_t groupsCountY = (cmaa2.sceneTonemappedUav.h() + threadGroupSize - 1) / threadGroupSize;
+
+    cmd.dispatch(groupsCountX, groupsCountY, 1);
```

Comment:
You don't need to calculate workgroup count manually, there is: `cmd.dispathThreads`.

Also `Y` and `Z` axis are `1` by default, so no need to specify `Z` in 2 dimensional workload.

> const uint32_t threadGroupSize = 8;

Also, if really needed there is `ComputePipeline::workGroupSize`